### DISCUSSION TITLE
twinBASIC BETA 68

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#968] [Releases]
- added: new Shapes of vbShapeArrowLeft/vbShapeArrowRight/vbShapeArrowUp/vbShapeArrowDown, with VariationA/B to control arrow proportions
- added: new Shape of vbShapeStar with VariationA/B/C properties allowing for many star-like shape variations
- added: Line control
- fixed: centering of Shape control for shapes of Circle/Square/RoundedSquare
- fixed: keypreview timing [ https://github.com/WaynePhillipsEA/twinbasic/issues/960 ]
- improved: form designer and runtime now supports container-painted controls (initially Shape and Line)
- fixed: (regression) Form.Unload event not firing, causing IDE window to not ask to save dirty project before closing
- fixed: 'Duplicate' option in form designer now ensures container controls are duplicated onto the form, not inside themselves [ https://github.com/WaynePhillipsEA/twinbasic/issues/958 ]

Note: container-painted controls (currently Line and Shape) are not supported inside ActiveX containers just yet (e.g Krools FrameW, SSTAB etc).
[twinBASIC_IDE_BETA_68.zip](https://github.com/WaynePhillipsEA/twinbasic/files/9046050/twinBASIC_IDE_BETA_68.zip)

<hr /><em>This discussion was created from the release <a href='https://github.com/WaynePhillipsEA/twinbasic/releases/tag/beta-68'>twinBASIC BETA 68</a>.</em>

**Comment by ScottRapsey** [comment]
I downloaded the zip from https://github.com/WaynePhillipsEA/twinbasic/releases/download/beta-68/twinBASIC_IDE_BETA_68.zip . Extracted it and then Windows 10 Virus & Threat Protection detected Trojan:Win32/Wacatac.B!ml in the twinBASIC.exe .
YMMV, I'm just letting you know what happened for me.

**Reply by WaynePhillipsEA** [reply]
Thanks for the report Scott.  Similar happened with BETA 67 (#959).  It is a false positive due to the release being so new.   I'm waiting on a code signing certificate being issued in the next day or two, which should help reduce these problems in the long term.

**Reply by ScottRapsey** [reply]
I took a real quick look before posting but I didn't find that discussion otherwise I would have just added to that one, sorry about that.
Thanks for the prompt response Wayne.

**Reply by WaynePhillipsEA** [reply]
No problem at all Scott, thanks
